### ISSUE TITLE
Console Example not working on ESP32-C6 (IDFGH-14116)

### ISSUE THREAD

**Issue by UlissesUy** [espressif/esp-idf#14922]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### General issue report

I compiled the example that illustrates the use of the console component ([esp-idf](https://github.com/espressif/esp-idf/tree/master)/[examples](https://github.com/espressif/esp-idf/tree/master/examples)/[system](https://github.com/espressif/esp-idf/tree/master/examples/system)/[console](https://github.com/espressif/esp-idf/tree/master/examples/system/console)/advanced/) to create an interactive shell on an ESP chip. It compiles without problem and loads (flash) to the microcontroller without problem. But when I try to connect with idf.py monitor gets stuck. When I connect with Putty it doesn't show anything. I'm using ESP-IDF v5.3.1. and ESP-IDE development environment. 
I've already searched and tried possible solutions but everything is still the same. Any ideas or suggestions? Thank you very much.
This is the message I get with idf.py:
idf.py monitor
Executing action: monitor
Serial port COM8
Connecting...
Detecting chip type... ESP32-C6
Running idf_monitor in directory C:\Users\Consola\myCLI\advanced
Executing "C:\Espressif\python_env\idf5.3_py3.11_env\Scripts\python.exe C:\Espressif\frameworks\esp-idf-v5.3.1\tools/idf_monitor.py -p COM8 -b 115200 --toolchain-prefix riscv32-esp-elf- --target esp32c6 --revision 0 --decode-panic backtrace C:\Users\Consola\myCLI\advanced\build\console.elf --force-color -m 'C:\Espressif\python_env\idf5.3_py3.11_env\Scripts\python.exe' 'C:\Espressif\frameworks\esp-idf-v5.3.1\tools\idf.py'"...
--- Warning: GDB cannot open serial ports accessed as COMx
--- Using \\.\COM8 instead...
--- esp-idf-monitor 1.5.0 on \\.\COM8 115200
--- Quit: Ctrl+] | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H
ESP-ROM:esp32c6-20220919
Build:Sep 19 2022
rst:0x15 (USB_UART_HPSYS),boot:0x18 (SPI_FAST_FLASH_BOOT)
Saved PC:0x40022ef0
--- 0x40022ef0: uart_rx_readbuff in ROM

SPIWP:0xee
mode:DIO, clock div:2
load:0x40875720,len:0x1338
load:0x4086c110,len:0xc70
load:0x4086e610,len:0x2c14
entry 0x4086c110
I (228) cpu_start: Unicore app
I (237) cpu_start: Pro cpu start user code
I (237) cpu_start: cpu freq: 160000000 Hz
I (238) app_init: Application information:
I (240) app_init: Project name:     console
I (245) app_init: App version:      1
I (249) app_init: Compile time:     Nov 22 2024 10:30:12
I (255) app_init: ELF file SHA256:  b575dabdc...
I (260) app_init: ESP-IDF:          HEAD-HASH-NOTFOUND
I (266) efuse_init: Min chip rev:     v0.0
I (271) efuse_init: Max chip rev:     v0.99
I (276) efuse_init: Chip rev:         v0.0
I (281) heap_init: Initializing. RAM available for dynamic allocation:
I (288) heap_init: At 408253A0 len 00057270 (348 KiB): RAM
I (294) heap_init: At 4087C610 len 00002F54 (11 KiB): RAM
I (300) heap_init: At 50000064 len 00003F84 (15 KiB): RTCRAM
I (307) spi_flash: detected chip: generic
I (311) spi_flash: flash io: dio
I (315) sleep: Configure to isolate all GPIO pins in sleep state
I (322) sleep: Enable automatic switching of GPIO sleep configuration

**Comment by peter-marcisovsky** [comment]
Hi @UlissesUy 
Which console output did you select in menuconfig before build? And which port are you monitoring? Also, which devkit are you using?

It looks like, you have flashed the console example for USB-UART console and monitoring JTAG port.

**Comment by UlissesUy** [comment]
Hi Peter, thanks for your reply. 
I am using FireBeetle 2 ESP32-C6 IoT Development Board. It has only one USB port. In menuconfig I used UART number 0. All settings are default.  
Thanks a lot.

**Comment by peter-marcisovsky** [comment]
Looking at the dev board you are using, it seems there is no usb-uart converter, so just select JTAG console in menuconfig, and the example shall work just fine.

**Comment by UlissesUy** [comment]
Hi Peter, thanks a lot for your reply. 
I'm using ESP-IDF V5.3.1. In menuconfig I can't find exactly the option “JTAG console”, I guess you mean the option: **Component config -> ESP System Setting -> Channel for console output (USB Serial/JTAG Controller)**. When enabling this option and compiling I get the following error:

C:/Users/Consola/myCLI/advanced/main/console_example_main.c:44:2: error: #error This example is incompatible with USB serial JTAG console.
   44 | #error This example is incompatible with USB serial JTAG console.
      |  ^~~~~
console_example_main.c: In function 'initialize_console':
C:/Users/Consola/myCLI/advanced/main/console_example_main.c:102:26: error: 'CONFIG_ESP_CONSOLE_UART_BAUDRATE' undeclared (first use in this function); did you mean 'CONFIG_ESP_CONSOLE_UART_NUM'?
  102 |             .baud_rate = CONFIG_ESP_CONSOLE_UART_BAUDRATE,
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

I don't understand why this type of port is not supported. I wanted to implement the console because it seems to me a good way to do an initial configuration for the device, for example, connect to the WiFi network or configure basic parameters. Just like the CLI of a router. 

Thank you very much.

**Comment by peter-marcisovsky** [comment]
Oh sorry, my bad.. I did console example refactoring some time ago and I just noticed you are using `IDF 5.3.x`. Could you try using the [basic console example](https://github.com/espressif/esp-idf/tree/release/v5.3/examples/system/console/basic) please?

If I remember correctly, there are 2 advanced console examples: one is `advanced`, this one uses UART interface, the other one is `advanced_usb_cdc` and uses `USB_OTG` peripheral (present only on esp32-s2 and esp32-s3). Thus, in IDF 5.3, there is no advanced console which uses JTAG interface.

The basic console supports all 3 interfaces (UART, USB_OTG and JTAG).

But.. starting in `IDF 5.4` , as you can check [here](https://github.com/espressif/esp-idf/tree/release/v5.4/examples/system/console) the two advanced console examples were merged into one and a JTAG support for advanced console was added.

To sum up.. 
- You can stay at `IDF 5.3.x` and use `basic` console example which supports JTAG interface
- Or you can update to `IDF 5.4.x` and use both, the `basic` and the `advanced` console examples and use JTAG interface

**Comment by UlissesUy** [comment]
Hi Peter,
OK. I'll try the basic console example.
Thanks a lot.

**Comment by UlissesUy** [comment]
Hi Peter,

I tried the basic console example and it works perfectly! Thank you so much for all your help.

Best regards!

**Comment by UlissesUy** [comment]
Iit works perfectly!
